### PR TITLE
Theme builder

### PR DESCRIPTION
**Pull request by Isha-Sharma**
#### Description

Fundwave component for setting theme on app.

#### Resolutions

- fixes getfundwave/discussions#187

### CODE REVIEW

File: showcase/fw-theme-builder/index.html [hunk 1]
```diff
@@ -0,0 +1,108 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta name="viewport" content="width=device-width,initial-scale=1.0">
+        <link rel="stylesheet" href="./style.css"/>
+        <script type="module" src="../../packages/fw-theme-builder/dist/fw-theme-builder/src/fw-theme-builder.js"></script>
+        <script>
+            window.addEventListener("color-change", (e) => {
+                console.log("Color:", e);
```
Comment (JRavi2) [inline showcase/fw-theme-builder/index.html]: Remove debug logs

File: packages/fw-theme-builder/readme.md
Comment (JRavi2) [inline packages/fw-theme-builder/readme.md:1]: Please add installation, usage instructions, example usage etc.

File: packages/fw-theme-builder/src/fw-font-pick.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
+import { html, css, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+
+@customElement("fw-font-pick")
+class FwFontPick extends LitElement {
+  @property({type : String})
+  label = "";
+
+  @property()
+  options: any;
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/fw-font-pick.ts]: Should avoid using `any`, better to define a type for options.
Also for any other places we're using `any`.

File: packages/fw-theme-builder/src/fw-size-pick.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+@customElement("fw-size-pick")
+class FwSizePick extends LitElement {
+  @property() 
+  label? : string;
+
+  @property()
+  value = "";
+
+  handleChange(e : any) {
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/fw-size-pick.ts]:
```suggestion
  handleChange(e : InputEvent) {
```

File: packages/fw-theme-builder/src/models.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  },
+  {
+    name : "Tajawal",
+    url: 'https://fonts.gstatic.com/s/tajawal/v9/Iurf6YBj_oCad4k1l8KiHrFpiQ.woff2',
+    style: "'Tajawal', sans-serif",
+  },
+]
+
+
+export const initialthemenew : any = {
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/models.ts]: `initialthemenew` -> `defaultTheme`

Also, no need for `any`.

File: packages/fw-theme-builder/src/models.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
+export const fontoptions = [
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/models.ts]: `fontoptions` -> `fontOptions`

File: packages/fw-theme-builder/src/fw-theme-builder.ts [hunk 6]
```diff
@@ -0,0 +1,313 @@
+import { html, css, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import "./fw-color-pick";
+import "./fw-size-pick";
+import "./fw-font-pick";
+import { fontoptions, initialthemenew } from "./models";
+
+@customElement("fw-theme-builder")
+class FwThemeBuilder extends LitElement {
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/fw-theme-builder.ts]: Should also export the classes

File: packages/fw-theme-builder/src/fw-color-pick.ts [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  const hexValues = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+
+  const red = hexValues ? parseInt(hexValues[1], 16) : 0;
+  const green = hexValues ? parseInt(hexValues[2], 16) : 0;
+  const blue = hexValues ? parseInt(hexValues[3], 16) : 0;
+
+  return { red, green, blue };
+}
+
+function rgbToHex(r: number, g: number, b: number) {
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/fw-color-pick.ts]: Can remove if we don't intend to use this

File: packages/fw-theme-builder/src/fw-theme-builder.ts [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+    }
+    .section-heading {
+      font-weight: 300;
+      margin: 1rem 0rem 0.25rem 0rem;
+      border-bottom: 1px solid #1b1b1b3b;
+      padding-bottom: 0.2rem;
+    }
+  `;
+
+  sectionChangeHandler(e: any, s: string) {
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/fw-theme-builder.ts]: Can remove the first param if we don't intend to use it

File: packages/fw-theme-builder/src/fw-theme-builder.ts [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+            (section) => html` <button
+              part="theme-button"
+              @click=${(e: any) => this.sectionChangeHandler(e, `${section}`)}
+            >
+              ${section}
+            </button>`
+          )}
+        </div>`;
+        break;
+      case "Colors":
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/fw-theme-builder.ts]: Better to keep these all uppercase or all lowercase. Even better to have them as enums.

File: packages/fw-theme-builder/src/fw-theme-builder.ts [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+    return html` <fw-font-pick
+      exportparts="font-container, font-label, font-button, font-dropdown-container, font-dropdown-option, font-dropdown-selected"
+      .label=${font}
+      @value-changed=${(e: any) => {
+        let detail = {
+          type: font,
+          value: e.detail.value,
+        };
+        this.shadowRoot?.dispatchEvent(
+          new CustomEvent("font-change", {
```
Comment (JRavi2) [inline packages/fw-theme-builder/src/fw-theme-builder.ts]:
```suggestion
          new CustomEvent("font-changed", {
```
Same for other events

**Review by JRavi2 — DISMISSED** [review]
Got some minor feedback. Rest lgtm.

File: packages/fw-theme-builder/readme.md
Comment (Isha-Sharma) [inline packages/fw-theme-builder/readme.md:1]: Please verify if all properties are correct here.

File: packages/fw-theme-builder/src/fw-theme-builder.ts [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+      width: 3.5rem;
+      user-select: none;
+      text-align: center;
+      border-radius: 4px;
+      font-family: "DM Sans", sans-serif;
+      padding: 0.5rem 0.5rem;
+      box-shadow: #1b1b1b3b 0px 4px 2px;
+    }
+    .save-btn {
+      background-color: #aaf16f;
```
Comment (Isha-Sharma) [inline packages/fw-theme-builder/src/fw-theme-builder.ts]: Do we use these?

If so, use css variables.

Can also verify if there's any redundant css or methods.

**Review by paras-fundwave — APPROVED** [review]

**Pull request merged**